napari/utils/theme.py

```diff
@@ -12,6 +12,12 @@
     use_gradients = False
 
 
+def __getattr__(attr):
+    if attr == "palettes":
+        return _themes
+    raise AttributeError
```

Comment:
Why not use `get_theme` and `register_theme` like in the [new_theme.py example](https://github.com/napari/napari/blob/master/examples/new_theme.py)? (And `available_themes`.) Happy to include this temporarily for backwards-compatibility, but it should be deprecated to use this. (And it should return a copy in either case.)

Reply:
I do not want a copy. I want to globally overwritten canvas background to have a black background in all themes. Multichannel confocal data bet looks if the background is black and additive blending for each layer. Even if the user decides to toggle the theme. 

I also need to compile a stylesheet to use it globally. 

Here is the current implementation:
https://github.com/4DNucleome/PartSeg/blob/7d386edd0fc00ec98caedda5ecce0e605665f1ef/package/PartSeg/common_backend/base_settings.py#L255-L259

Maybe it could be done better because the code for this was written when I start embedding the viewer.

Reply:
Well, I just checked and [`register_theme`](https://github.com/napari/napari/blob/a0b38907487d8bd32d2a7300bcbbecfcd5b00b40/napari/utils/theme.py#L149-L159) does not check whether you are overwriting an existing theme, so I think that would be the canonical way of doing things now:

- grab the theme (always a copy)
- edit the bits you want (in this case, the background)
- register the new theme under the existing name

Even if in the future we decide to add a check for overwriting, I think we can easily add a `force=True` kwarg that allows one to always overwrite.

Reply:
btw, I do agree with you that medium-term we probably want to avoid the canvas background from being managed by the theme, as it is something more fundamental to the visualisation itself. (The theme instead having to do with the look of the UI elements.)

Reply:
> Even if in the future we decide to add a check for overwriting, I think we can easily add a `force=True` kwarg that allows one to always overwrite.

I do not like this. Adding such a parameter will force immediate code update. I would rather decide to add `force` before `0.4.3` release or decide that overwrite without `force` will work in the future.

Reply:
To clarify, I'm not planning on doing this, only saying, we still have that option if overwriting-by-default causes some unforeseen chaos.